packages/hdwallet-walletconnect-bridge/src/bridge.test.ts

```diff
@@ -0,0 +1,155 @@
+import * as core from "@shapeshiftoss/hdwallet-core";
+import WalletConnect from "@walletconnect/client";
+
+import { HDWalletWCBridge } from "./bridge";
+
+describe("HDWalletWCBridge", () => {
+  let bridge: HDWalletWCBridge;
+  let connector: Partial<WalletConnect>;
+
+  beforeEach(async () => {
+    const wallet: Partial<core.ETHWallet> = {
+      _supportsETH: true,
+      ethGetAddress: () => Promise.resolve("test address"),
+      ethSignTx: () =>
+        Promise.resolve({
+          v: 0,
+          r: "r",
+          s: "s",
+          serialized: "serialized signature",
+        }),
+      ethSendTx: () => Promise.resolve({ hash: "tx hash" }),
+      ethSignMessage: () =>
+        Promise.resolve({
+          address: "test address",
+          signature: "test signature",
+        }),
+      ethVerifyMessage: () => Promise.reject(new Error("not implemented")),
+    };
+    connector = {
+      on: jest.fn(),
+      createSession: jest.fn(),
+      approveSession: jest.fn(),
+      approveRequest: jest.fn(),
+      rejectRequest: jest.fn(),
+    };
+    bridge = new HDWalletWCBridge(wallet as core.ETHWallet, connector as WalletConnect);
+  });
+
+  it("should subscribe to events when connecting", async () => {
+    await bridge.connect();
+
+    expect(connector.createSession).toHaveBeenCalledTimes(1);
+    expect(connector.on).toHaveBeenCalledWith("session_request", expect.any(Function));
+    expect(connector.on).toHaveBeenCalledWith("session_update", expect.any(Function));
+    expect(connector.on).toHaveBeenCalledWith("call_request", expect.any(Function));
+  });
+
+  describe("on session_request", () => {
+    it("should approve session with current chain and account", async () => {
+      await bridge._onSessionRequest(null, { params: [{ chainId: 123 }] });
+      expect(connector.approveSession).toHaveBeenCalledWith({
+        chainId: 123,
+        accounts: ["test address"],
+      });
+    });
+
+    it("should default to rinkeby chain", async () => {
```

Comment:
Why does WalletConnect default to Rinkeby instead of mainnet? Do we have to make sure to set the chainId manually after initialization?

Reply:
It's me defaulting to Rinkeby: https://github.com/shapeshift/hdwallet/pull/553/files#diff-8df98eb3ac395c391f2d7e2f84e1151d5f79595b55ae48d6e37cba5eab3f6d74R53
We could change it to mainnet. WC needs a chainId to approve a session, but doesn't necessarily provide what chain ID when connecting, so we're defaulting it